tests/envphys/solar/test_toporad.py

```diff
@@ -0,0 +1,226 @@
+import numpy as np
+import pandas as pd
+import os
+
+from topocalc.shade import shade
+
+from smrf.data import loadTopo
+from smrf.envphys.sunang import sunang
+from smrf.envphys.solar import toporad, irradiance
+from smrf.envphys.albedo import albedo
+from smrf.utils import utils
+from tests.smrf_test_case_lakes import SMRFTestCaseLakes
+
+
+class TestToporad(SMRFTestCaseLakes):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        date_time = pd.to_datetime('2/15/1990 20:30')
+        cls.date_time = date_time.tz_localize('UTC')
+
+        cls.tau_elevation = 100.0
+        cls.tau = 0.2
+        cls.omega = 0.85
+        cls.scattering_factor = 0.3
+        cls.surface_albedo = 0.5
+        cls.solar_irradiance = irradiance.direct_solar_irradiance(
+            cls.date_time, w=[0.28, 2.8])
+
+        topo_config = {
+            'filename': os.path.join(cls.test_dir, 'topo/topo.nc'),
+            'northern_hemisphere': True,
+            'gradient_method': 'gradient_d8',
+            'sky_view_factor_angles': 72
+        }
+        cls.topo = loadTopo.Topo(
+            topo_config,
+            tempDir=os.path.join(cls.test_dir, 'output')
+        )
+        cls.dem = cls.topo.dem
+
+        # inputs for toporad and stoporad
+        cls.cosz, cls.azimuth, rad_vec = sunang(
+            cls.date_time,
+            cls.topo.basin_lat,
+            cls.topo.basin_long)
+
+        cls.elevrad = toporad.Elevrad(
+            cls.dem,
+            cls.solar_irradiance,
+            cls.cosz)
+
+        cls.illum_ang = shade(
+            cls.topo.sin_slope,
+            cls.topo.aspect,
+            cls.azimuth,
+            cls.cosz)
+
+    def setup(self):
+        super().setup()
+
+        self.assertTrue(self.tau_elevation == 100.0)
+        self.assertTrue(self.tau == 0.2)
+        self.assertTrue(self.omega == 0.85)
+        self.assertTrue(self.scattering_factor == 0.3)
+        self.assertTrue(self.surface_albedo == 0.5)
```

Comment:
Why is this tested with each call to `test_XXX`?

Reply:
The tests were passing individually but not when the whole class was ran so I was making sure things weren't changing. Turned out that a .copy() was needed in stoporad. I've removed.